FirebaseAuth/Sources/User/FIRUser.m

```diff
@@ -1129,8 +1129,6 @@ - (void)linkWithEmailPassword:(FIREmailPasswordAuthCredential *)credential
                         callInMainThreadWithAuthDataResultAndError(completion, nil, keychainError);
                         return;
                       }
-                      [self signOutIfTokenIsInvalidWithError:error];
-                      callInMainThreadWithAuthDataResultAndError(completion, authResult, nil);
```

Comment:
Isn't this the success callback to the caller?

Reply:
Yes, correct. Tested e2e and it should work fine now.